### PR TITLE
detect config is readonly, adalight and other stuff

### PR DESCRIPTION
**Pull request by redPanther**
**1.** Tell us something about your changes.
- update adalight sketch (this won't be the last)
- update lightberry sketches to latest upstream
- add tpm2 sketch. mainly for dev's so we can test tpm2
- tune ada driver, set delayAfterConnect default to 1.5s because some arduino (e.g. mega r3) needs this
- update compilehowto (windows disclaimer)
- some refactoring in main cmakelists + preparation for windows compile
- set priority min/max for grabber/network services - prevent colliding prios between webui/background stuff and grabbers/net services
- add check if config is writable. TODO do something usefull in webui
- fix webui can't write led config

**2.** If this changes affect the .conf file. Please provide the changed section
no

**3.** Reference an issue (optional)
#330 #331 #292

### CODE REVIEW

File: assets/webconfig/js/content_leds.js [hunk 1]
```diff
@@ -472,11 +472,8 @@ $(document).ready(function() {
 	
 	// ------------------------------------------------------------------
 	$("#leds_custom_save").off().on("click", function() {
-		function createLedConfig(){
-			var string = '{"leds" :'+$("#ledconfig").val()+'}';
-		}
 		if (validateText())
-			requestWriteConfig(JSON.parse(createLedConfig()));
+			requestWriteConfig(JSON.parse('{"leds" :'+$("#ledconfig").val()+'}'));
```
Comment (brindosch) [inline assets/webconfig/js/content_leds.js]: uhm, of course. That was my intention. For a unknown reason i stopped in between :)

**Comment by brindosch** [comment]
Nice!
Will do the ui stuff.

**Comment by redPanther** [comment]
> Will do the ui stuff.
I was  hoping you will do. BTW I tested webui on my x86 and my pi1 both can write the config. (tried my standard config with 116 leds)

**Comment by brindosch** [comment]
PLATFORM: rpi

ENABLE_FB = OFF
ENABLE_SPIDEV   = OFF
ENABLE_V4L2 = OFF

Hm, no :)

File: src/hyperiond/hyperiond.cpp [hunk 2]
```diff
@@ -72,9 +72,9 @@ HyperionDaemon::HyperionDaemon(QString configFile, QObject *parent)
 		WarningIf(_qconfig.contains("logger"), Logger::getInstance("LOGGER"), "Logger settings overriden by command line argument");
```
Comment (brindosch) [inline src/hyperiond/hyperiond.cpp]: overridden :)

**Comment by brindosch** [comment]
The notifications seems not to work for me. I edited the write access and restartet always. But it still hangs in writeable : true

-r--r--r-- 1 osmc osmc 20553 Dec 14 00:23 hyperion.config.json

**Comment by b1rdhous3** [comment]
Ich hab den Adalight-Treiber nun mal mit Lightberry HD und dem USB Connector getestet. 
Läuft alles! :)

File: src/hyperiond/hyperiond.cpp (hunk 2 above)
Comment (redPanther) [inline src/hyperiond/hyperiond.cpp]: we should make a text typo hunt ;-) I'm sure we have more. Perhaps a english native can help ;-)

**Comment by redPanther** [comment]
@brindosh you are running as root, than all files are writeable ragardless of the permissions. I use hyperion as user, then this check works.

**Comment by brindosch** [comment]
Ah, wonder! I will try as non root again :)
I am still slightly confused, as i had the hope this was the reason why i can't save configs.
Will test futher.

**Comment by redPanther** [comment]
Saving: I figured out that I can save only once per session. That means Ihave to reload the ui. Now I can reproduce and figure out waht's the reason.

Btw I successfully created my special led layout with 96 ws2812 leds + 4 ikea dioder stripes with the webui. What I missed is an option to append generated config to current, instead of replaceing it.

**Comment by brindosch** [comment]
> What I missed is an option to append generated config to current, instead of replaceing it.

append a led layout to the current? I can't imagine something different.
I thought about a backup/restore function. but we need a version key in config that is maintained from Hyperion itself to keep old configs away

**Pull request merged**